enterprise_access/apps/api/v1/views/customer_billing.py

```diff
@@ -338,8 +337,8 @@ def partial_update(self, request, *args, **kwargs):
 
         # Check if state is being updated
         new_state = request.data.get('state')
-        if new_state and new_state != instance.state:
-            if not self._is_valid_state_transition(instance.state, new_state):
+        if new_state:
+            if not CheckoutIntent.is_valid_state_transition(instance.state, new_state):
```

Comment:
[nit] Could this be a single conditional with an `and` operator? As long as new_state always gets checked first, it should always short circuit the logic before passing the value against the `CheckoutIntent`'s, `is_valid_state_transition` method.